src/celeritas/em/distribution/WentzelDistribution.hh

```diff
@@ -0,0 +1,293 @@
+    inline CELER_FUNCTION real_type mom_transfer_sq(real_type cos_t) const;
+
+    //! Momentum coefficient used in the flat nuclear form factor model
+    CELER_CONSTEXPR_FUNCTION real_type flat_coeff() const;
+
+    //! Momentum prefactor used in exponential and gaussian form factors
+    inline CELER_FUNCTION real_type nuclear_form_prefactor() const;
+
+    //! Samples the scattered polar angle based on the maximum scattering
+    //! angle and screening coefficient.
+    template<class Engine>
+    inline CELER_FUNCTION real_type sample_cos_t(real_type cos_t_max,
+                                                 Engine& rng) const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with state and data from WentzelInteractor
+ */
+CELER_FUNCTION
+WentzelDistribution::WentzelDistribution(ParticleTrackView const& particle,
+                                         IsotopeView const& target,
+                                         WentzelElementData const& element_data,
+                                         real_type cutoff_energy,
+                                         WentzelRef const& data)
+    : data_(data)
+    , particle_(particle)
+    , target_(target)
+    , element_data_(element_data)
+    , wentzel_elec_ratio(particle, target.atomic_number(), data, cutoff_energy)
+{
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Samples the final direction of the interaction. This direction is in
+ * the frame where the incident particle's momentum is oriented along the
+ * z-axis, so it's final direction in the lab frame will need to be rotated.
+ */
+template<class Engine>
+CELER_FUNCTION Real3 WentzelDistribution::operator()(Engine& rng) const
+{
+    real_type cos_theta = 1;
+    if (BernoulliDistribution(wentzel_elec_ratio())(rng))
+    {
+        // Scattered off of electrons
+        cos_theta = sample_cos_t(wentzel_elec_ratio.cos_t_max_elec(), rng);
+    }
+    else
+    {
+        // Scattered off of nucleus
+        cos_theta = sample_cos_t(-1, rng);
+
+        // Calculate rejection for fake scattering
+        // TODO: Reference?
+        real_type mott_coeff
+            = 1 + real_type(1e-4) * ipow<2>(target_.atomic_number().get());
+        MottRatioCalculator mott_xsec(element_data_,
+                                      std::sqrt(particle_.beta_sq()));
+        real_type g_rej = mott_xsec(cos_theta)
+                          * ipow<2>(calculate_form_factor(cos_theta))
+                          / mott_coeff;
+
+        if (!BernoulliDistribution(g_rej)(rng))
+        {
+            return {0, 0, 1};
+        }
+    }
+
+    // Calculate scattered vector assuming azimuthal angle is isotropic
+    UniformRealDistribution<real_type> sample_phi(0, 2 * constants::pi);
+    return from_spherical(cos_theta, sample_phi(rng));
```

Comment:
Since your distribution is just sampling the cosine polar angle, maybe change the signature to return just a scalar, and do the polar angle sampling downstream? That'll simplify the code where you extract the Z component of the returned result.